### ISSUE TITLE
Add closeness centrality function

### ISSUE THREAD

**Issue by jhollway** [jhollway/migraph#29]
**[Original report](https://bitbucket.org/jhollway/roctopus/issue/29) by James Hollway (Bitbucket: [jhollway](https://bitbucket.org/jhollway), GitHub: [jhollway](https://github.com/jhollway)).**

----------------------------------------

**Comment by henriquesposito** [comment]
I have added a function that builds upon `tidygraph::centrality_closeness()` but it does not yet pass the tests. I run into two issue here.

```
centrality_closeness <- function (weights = NULL, mode = "out", normalized = FALSE, cutoff = NULL){
  tidygraph:::expect_nodes()
  graph <- .G()
  weights <- enquo(weights)
  weights <- rlang::eval_tidy(weights, .E())
  if (is.null(weights)) {
    weights <- NA
  }
  if (is_bipartite(graph) & normalized){
    closeness <- closeness(graph = graph, v = V(graph), mode = mode, cutoff = cutoff)
    other_set_size <- ifelse(V(graph)$type, sum(!V(graph)$type), sum(V(graph)$type))
    set_size <- sum(V(graph)$type)
    closeness/(other_set_size+(2*set_size)-2)
    } else {
      if (is.null(cutoff)) {
        closeness(graph = graph, vids = V(graph), mode = mode,
                  weights = weights, normalized = normalized)
      } else {
        estimate_closeness(graph = graph, vids = V(graph), mode = mode, 
                           cutoff = cutoff, weights = weights, normalized = normalized)
      }
    }
} 
```

First, we get non-normalized betweenness scores for the first two tests and the results displayed in the Borgatti and Everrett (1997) for one mode closeness, used to build the tests, are actually normalized one mode. This could be fixed by changing the test scores to non-normalized ones. Would that be okay? That would mean that the table is not a reference for one part of the test. 

Second, the latter tests for two-mode, do not calculate because we get an error that says some of the arguments are unused. I have tried to change the parameters or the values of the parameters, but I cannot get the function to work. 

Below is the output of the test failure. 

```
> test_that("two mode closeness centrality calculated correctly",{
+   expect_equal(unname(with_graph(as_tbl_graph(southern_women), 
+                                  roctopus::centrality_closeness())[1:5]), 
+                c(51.67,46.97,51.67,46.97,38.75))
+   expect_equal(unname(with_graph(as_tbl_graph(southern_women), 
+                                  roctopus::centrality_closeness())[28:32]), 
+                c(39.74,37.80,40.79,37.80,37.00))
+   expect_equal(unname(round(with_graph(as_tbl_graph(southern_women), 
+                                        roctopus::centrality_closeness(normalized = T))[1:5],4)), 
+                c(80.00, 72.73, 80.00, 72.73, 60.00))
+   expect_equal(unname(round(with_graph(as_tbl_graph(southern_women), 
+                                        roctopus::centrality_closeness(normalized = T))[28:32],4)), 
+                c(56.42, 53.66, 57.89, 53.66, 53.66))
+ })
-- Failure (Line 2): two mode closeness centrality calculated correctly --------------------------------------------------------------------------------------
unname(with_graph(as_tbl_graph(southern_women), roctopus::centrality_closeness())[1:5]) not equal to c(51.67, 46.97, 51.67, 46.97, 38.75).
5/5 mismatches (average diff: 47.2)
[1] 0.0167 - 51.7 == -51.7
[2] 0.0152 - 47.0 == -47.0
[3] 0.0167 - 51.7 == -51.7
[4] 0.0152 - 47.0 == -47.0
[5] 0.0125 - 38.8 == -38.7

-- Failure (Line 5): two mode closeness centrality calculated correctly --------------------------------------------------------------------------------------
unname(with_graph(as_tbl_graph(southern_women), roctopus::centrality_closeness())[28:32]) not equal to c(39.74, 37.8, 40.79, 37.8, 37).
5/5 mismatches (average diff: 38.6)
[1] 0.0128 - 39.7 == -39.7
[2] 0.0122 - 37.8 == -37.8
[3] 0.0132 - 40.8 == -40.8
[4] 0.0122 - 37.8 == -37.8
[5] 0.0122 - 37.0 == -37.0

-- Error (Line 8): two mode closeness centrality calculated correctly ----------------------------------------------------------------------------------------
Error: unused argument (cutoff = cutoff)
Backtrace:
 1. testthat::expect_equal(...)
 6. roctopus::centrality_closeness(normalized = T)
```

@jhollway Do you understand what I am doing wrong here?

**Comment by jhollway** [comment]
Closed in #71
